core/tx_pool.go

```diff
@@ -363,7 +366,10 @@ func (pool *TxPool) loop() {
 				// Any non-locals old enough should be removed
 				if time.Since(pool.beats[addr]) > pool.config.Lifetime {
 					for _, tx := range pool.queue[addr].Flatten() {
-						pool.removeTx(tx.Hash(), true)
+						if time.Since(pool.queued_ts[tx.Hash()]) > pool.config.Lifetime {
```

Comment:
If we get a mismatch between what's in `pool.queued_tx` and other pools, the `pool.queued_ts[tx.Hash()]` will return `0`, and flush out the tx. I suppose that is indeed the correct way to handle that error. 
For testing though, it might be interesting to make it `panic`, and run it for a few days with a `panic` in there, just to see that we don't have any inconsistencies.

Reply:
I just updated this to panic, thanks!

Reply:
Sorry, I was a bit unclear. We can't merge it to `master` with a `panic` in there, it was more of a note-to-self (or whoever puts it on a prod server) to put a panic check there.

Reply:
I'll drop the last commit and rebase it on master, ok?